File: ios/Treddy/Services/DiscoveryUrl.swift
import Foundation

/// Pure mapping from a resolved mDNS record to the app's base server URL.
/// No platform imports beyond Foundation — testable on macOS or any
/// non-iOS host. scheme defaults to https (the Pi serves a self-signed
/// cert; iOS trusts it via TrustAllDelegate — see precor-9_3x-41a).
func discoveredBaseURL(host: String, port: Int, txt: [String: String]) -> String {
    let scheme = (txt["scheme"]?.isEmpty == false) ? txt["scheme"]! : "https"
    return "\(scheme)://\(host):\(port)"
}

File: kotlin/app/src/main/java/com/precor/treadmill/discovery/DiscoveryUrl.kt
package com.precor.treadmill.discovery

/**
 * Pure mapping from a resolved mDNS record to the app's base server URL.
 * No Android imports — unit-testable on the JVM. scheme defaults to https
 * (the Pi serves a self-signed cert; clients trust it — see precor-9_3x-41a).
 */
fun discoveredBaseUrl(host: String, port: Int, txt: Map<String, String>): String {
    val scheme = txt["scheme"]?.takeIf { it.isNotBlank() } ?: "https"
    return "$scheme://$host:$port"
}
